containers/src/Data/IntMap.hs

```diff
@@ -59,42 +48,6 @@
 
 module Data.IntMap
     ( module Data.IntMap.Lazy
-#ifdef __GLASGOW_HASKELL__
--- For GHC, we disable these, pending removal. For anything else,
--- we just don't define them at all.
-    , insertWith'
-    , insertWithKey'
-    , fold
-    , foldWithKey
-#endif
     ) where
 
 import Data.IntMap.Lazy
-
-#ifdef __GLASGOW_HASKELL__
-import Utils.Containers.Internal.TypeError
-
--- | This function is being removed and is no longer usable.
--- Use 'Data.IntMap.Strict.insertWith'
-insertWith' :: Whoops "Data.IntMap.insertWith' is gone. Use Data.IntMap.Strict.insertWith."
```

Comment:
While bumping lower bound is just a maintenance thing, the removal of functions is a breaking change.

Why you do them at the same time? Is it time for `containers-0.7`?

Reply:
@phadej I was just thinking in terms of getting rid of cruft. Are you opposed to the change?

Reply:
New idea! Let's delete those without a major release. I argue that's okay. The functions were totally disabled anyway, so the disruption of removing them is limited to potentially having to delete them from `hiding` clauses. This is no worse than the potential disruption of *adding* functions, which is allowed in minor releases.

Reply:
You should removed them in `containers-0.6`. Removal of symbols whenever they are usable/useful or not is a breaking change, period.

Reply:
Concrete and not too contrived example: Someone could still explicitly-re-export these symbols, removal would break that usage.